omnigibson/systems/macro_particle_system.py

```diff
@@ -112,79 +116,65 @@ def _create_particle_template(cls):
         """
         raise NotImplementedError()
 
-    @classmethod
-    def remove_all_particles(cls):
+    def remove_all_particles(self):
         # Use list explicitly to prevent mid-loop mutation of dict
-        for particle_name in tuple(cls.particles.keys()):
-            cls.remove_particle_by_name(name=particle_name)
+        for particle_name in tuple(self.particles.keys()):
+            self.remove_particle_by_name(name=particle_name)
 
-    @classmethod
-    def reset(cls):
+    def reset(self):
         # Call super first
         super().reset()
 
         # Reset the particle counter
-        cls._particle_counter = 0
+        self._particle_counter = 0
 
-    @classmethod
-    def _clear(cls):
+    def _clear(self):
         # Clear all internal state
-        og.sim.remove_object(cls._particle_template)
+        og.sim.remove_object(self._particle_template)
 
         super()._clear()
 
-        cls._particle_template = None
-        cls.particles = None
-        cls._color = None
+        self._particle_template = None
+        self.particles = dict()
```

Comment:
same here -- presumably after clear, initialize() must be called again, so it's good to maintain distinct variable states between the two